src/robotFactory.js

```diff
@@ -1,10 +1,86 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    const {
+      x = 0,
+      y = 0,
+    } = coords;
 
-class DeliveryDrone {}
+    this.coords = {
+      x: x,
+      y: y,
+    };
```

Comment:
```suggestion
    this.coords = {
      x,
      y
    };
```
Можна використовувати скорочений запис